version: "3.2"

services:
  adminer:
    container_name: adminer
    image: adminer:latest
    restart: always
    ports:
      - 8080:8080
    networks:
      - app-net
  course:
    container_name: course
    build:
      context: .
      dockerfile: ./build/Dockerfile
    restart: on-failure
    volumes:
      - ./bin:/home/app/bin
      - ./tests:/home/app/tests
      - ./seeds:/home/app/seeds
      - ./src:/home/app/src
      - ./manage.py:/home/app/manage.py
      - ./setup.py:/home/app/setup.py
    ports:
      - 20008:5000
    env_file:
      - env/.env.course
    healthcheck:
      test: curl --fail -s course:5000/ping || exit 1
      interval: 60s
      timeout: 5s
      retries: 5
    depends_on:
      - course_db
    networks:
      - app-net
  course_db:
    container_name: course_db
    image: postgres:9.6
    volumes:
      - db_data:/var/lib/postgresql/data/
    env_file:
      - env/.env.course_db
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U course"]
      interval: 10s
      timeout: 5s
      retries: 20
    networks:
      - app-net
volumes:
  db_data:
networks:
  app-net:
    driver: bridge